python_modules/dagster/dagster_tests/definitions_tests/decorators_tests/test_asset_decorator_with_check_specs.py

```diff
@@ -333,6 +328,48 @@ def asset_1_and_2():
         materialize(assets=[asset_1_and_2])
 
 
+def test_multi_asset_blocking_check_skips_downstream():
+    @multi_asset(
+        outs={"one": AssetOut("asset1"), "two": AssetOut("asset2")},
+        check_specs=[AssetCheckSpec("check1", asset=AssetKey(["asset1", "one"]), blocking=True)],
+    )
+    def asset_1_and_2():
+        yield Output(None, output_name="one")
+        yield Output(None, output_name="two")
+        yield AssetCheckResult(check_name="check1", passed=False)
+
+    @asset(deps=[AssetKey(["asset1", "one"])])
+    def asset3():
+        pass
+
+    @asset(deps=[AssetKey(["asset2", "two"])])
+    def asset4():
+        pass
+
+    result = materialize(assets=[asset_1_and_2, asset3, asset4], raise_on_error=False)
+    assert not result.success
+
+    # note: currently asset4 won't run because asset_1_and_2 failed, but ideally it would run
+    # because only the check on asset1 failed.
+    materialization_events = result.get_asset_materialization_events()
```

Comment:
this will be valuable to fix for dbt -> python especially

Reply:
@alangenfeld maybe we could chat about how to implement this? Currently when a blocking check fails, we just raise an exception to make the whole step fail and block downstream steps. Ideally in a multi_asset we could stop specific downstreams that depended on the check and let others run

Reply:
ya this is tricky, may be best to chat on a call.

Reply:
I'll reach out- shouldn't block this diff but will be important for dbt -> python asset dependencies